tssc/step_implementers/validate_environment_configuration/configlint.py

```diff
@@ -0,0 +1,231 @@
+        -rules rules.yml /home/user/tssc-working/deploy/file.yml
+
+**Example: Existing Rules File (minimal)**
+
+    version: 1
+    description: Rules for Kubernetes spec files
+    type: Kubernetes
+    files:
+      - "*.yml"
+    rules:
+    - id: TSSC_EXAMPLE
+      severity: FAILURE
+      message: Deployment must have istio
+      resource: Deployment
+      assertions:
+        - key: spec.template.metadata.annotations
+          op: contains
+          value: '"sidecar.istio.io/inject": false'
+
+Example: Results
+
+    'tssc-results': {
+        'validate-environment-configuration': {
+            'result': {
+                'success': True,
+                'message': 'config-lint step completed'
+            }
+            'options': {
+                'yml_path': '/home/user/tssc-working/file.yml'
+            }
+
+    }
+
+.. Note:: The configuration example above used the ConfiglintFromArgocd step to \
+provide the yml_path option.
+
+"""
+
+import os
+import sys
+import sh
+from tssc import StepImplementer
+
+DEFAULT_CONFIG = {
+    'rules': './config-lint.rules'
+}
+
+REQUIRED_CONFIG_KEYS = {
+}
+
+AUTHENTICATION_CONFIG = {
+}
+
+class Configlint(StepImplementer):
+    """
+    StepImplementer for the tag-source step for Config_lint.
+    """
+
+    @staticmethod
+    def step_implementer_config_defaults():
+        """
+        Getter for the StepImplementer's configuration defaults.
+
+        Notes
+        -----
+        These are the lowest precedence configuration values.
+
+        Returns
+        -------
+        dict
+            Default values to use for step configuration values.
+        """
+        return DEFAULT_CONFIG
+
+    @staticmethod
+    def required_runtime_step_config_keys():
+        """
+        Getter for step configuration keys that are required before running the step.
+
+        See Also
+        --------
+        _validate_runtime_step_config
+
+        Returns
+        -------
+        array_list
+            Array of configuration keys that are required before running the step.
+        """
+        return REQUIRED_CONFIG_KEYS
+
+    def _run_step(self):
+        """
+        Runs the TSSC step implemented by this StepImplementer.
+
+        Parameters
+        ----------
+        runtime_step_config : dict
+            Step configuration to use when the StepImplementer runs the step with all of the
+            various static, runtime, defaults, and environment configuration munged together.
+
+        Returns
+        -------
+        dict
+            Results of running this step.
+        """
+
+        # Find yml_path
+        yml_path = self.get_config_value('yml_path')
+        if yml_path is None:
+            try:
+                current_step_results = self.current_step_results()
+                if 'options' in current_step_results:
+                    if 'yml_path' in current_step_results['options']:
+                        yml_path = current_step_results['options']['yml_path']
```

Comment:
The enclosing try-catch can be removed by changing:
```
yml_path = current_step_results['options']['yml_path']
```
to
```
yml_path = current_step_results['options'].get('yml_path')
```

Reply:
changed and removed the try/catch.   
The reason for the try/catch was to catch an error if "current_step_results" failed; my unit test case was overzealous.